### ISSUE TITLE
1.16.5 How to disable leave msg after the kick msg?

### ISSUE THREAD

**Issue by nikitushu2** [kangarko/ChatControl#945]
Minecraft Version: 1.16.5
MySQL: true
BungeeCord: false

When players get kicked, the message is broadcasted. But at the same time, the message about their leaving is broadcasted as well, which is unnecessary in this case. How to disable the leave message (only after the kick msgs)?

**Comment by ElBananaa** [comment]
> Minecraft Version: 1.16.5
> MySQL: true
> BungeeCord: false
> 
> When players get kicked, the message is broadcasted. But at the same time, the message about their leaving is broadcasted as well, which is unnecessary in this case. How to disable the leave message (only after the kick msgs)?

Hey!
Unfortunately, I can't reproduce this issue. Which plugin handles kicks on your server?

In the plugin settings file, under the `Messages.Apply_On` part, try to remove the kick part.
This is how it should look after : 
```yaml
Messages:
 
  # What aspects of messages should be enabled?
  # Options: join, quit, kick, death, timed
  #
  # If you have another plugin handling join/quit etc. messages, remove them from list below.
  # If you want to hide those messages, leave them in Apply_On and empty your messages/ .rs files.
  Apply_On:
  - 'join'
  - 'quit'
  - 'death'
  - 'timed'
```
Let me know if it works. :)

**Comment by kangarko** [comment]
I'm afraid that would be impossible to catch through the native Bukkit API since there's no "quit reason" that a third party plugin can modify... 

<img width="543" alt="Screenshot 2021-05-02 at 20 05 32" src="https://user-images.githubusercontent.com/1918426/116822916-c50c9880-ab81-11eb-9698-0a9aacc81223.png">

Try removing the "kick" from Apply_On as Banana suggested, or ask the author of your plugin this suggestion: "Change your PlayerQuitEvent priority to LOW or LOWEST and set the quit reason to something I can recognize in chatcontrol such as "#silence" so that when it's chatcontrol's we can just ignore and not print any message"

**Comment by nikitushu2** [comment]
> 
> 
> > Minecraft Version: 1.16.5
> > MySQL: true
> > BungeeCord: false
> > When players get kicked, the message is broadcasted. But at the same time, the message about their leaving is broadcasted as well, which is unnecessary in this case. How to disable the leave message (only after the kick msgs)?
> 
> Hey!
> Unfortunately, I can't reproduce this issue. Which plugin handles kicks on your server?
> 
> In the plugin settings file, under the `Messages.Apply_On` part, try to remove the kick part.
> This is how it should look after :
> 
> ```yaml
> Messages:
>  
>   # What aspects of messages should be enabled?
>   # Options: join, quit, kick, death, timed
>   #
>   # If you have another plugin handling join/quit etc. messages, remove them from list below.
>   # If you want to hide those messages, leave them in Apply_On and empty your messages/ .rs files.
>   Apply_On:
>   - 'join'
>   - 'quit'
>   - 'death'
>   - 'timed'
> ```
> 
> Let me know if it works. :)

I tried this but it removed all the kick messages. And if I remove the quit message, then all quit messages disappear.

Just in case, what I am trying to achieve:
1. Quit messages remain.
2. Kick messages remain.
3. Quit messages appearing straight after kick messages should be removed. (because if a player is kicked, then there is no need for players to know that he left too, which takes extra space in the chat [logically kick assumes the leaving of a player too])

**Comment by nikitushu2** [comment]
> 
> 
> I'm afraid that would be impossible to catch through the native Bukkit API since there's no "quit reason" that a third party plugin can modify...
> 
> <img alt="Screenshot 2021-05-02 at 20 05 32" width="543" src="https://user-images.githubusercontent.com/1918426/116822916-c50c9880-ab81-11eb-9698-0a9aacc81223.png">
> 
> Try removing the "kick" from Apply_On as Banana suggested, or ask the author of your plugin this suggestion: "Change your PlayerQuitEvent priority to LOW or LOWEST and set the quit reason to something I can recognize in chatcontrol such as "#silence" so that when it's chatcontrol's we can just ignore and not print any message"

Alright, I'll ask them. Thanks

**Comment by nikitushu2** [comment]
> 
> 
> I'm afraid that would be impossible to catch through the native Bukkit API since there's no "quit reason" that a third party plugin can modify...
> 
> <img alt="Screenshot 2021-05-02 at 20 05 32" width="543" src="https://user-images.githubusercontent.com/1918426/116822916-c50c9880-ab81-11eb-9698-0a9aacc81223.png">
> 
> Try removing the "kick" from Apply_On as Banana suggested, or ask the author of your plugin this suggestion: "Change your PlayerQuitEvent priority to LOW or LOWEST and set the quit reason to something I can recognize in chatcontrol such as "#silence" so that when it's chatcontrol's we can just ignore and not print any message"

The plugin's dev (Spartan plugin, that kicks players) said:

Kicking a player and having no quit/leave message are two completely different features
It's something Spartan doesn't mess with and something you will need to resolve on your own

What to do then?

**Comment by ElBananaa** [comment]
> > I'm afraid that would be impossible to catch through the native Bukkit API since there's no "quit reason" that a third party plugin can modify...
> > <img alt="Screenshot 2021-05-02 at 20 05 32" width="543" src="https://user-images.githubusercontent.com/1918426/116822916-c50c9880-ab81-11eb-9698-0a9aacc81223.png">
> > Try removing the "kick" from Apply_On as Banana suggested, or ask the author of your plugin this suggestion: "Change your PlayerQuitEvent priority to LOW or LOWEST and set the quit reason to something I can recognize in chatcontrol such as "#silence" so that when it's chatcontrol's we can just ignore and not print any message"
> 
> The plugin's dev (Spartan plugin, that kicks players) said:
> 
> Kicking a player and having no quit/leave message are two completely different features
> It's something Spartan doesn't mess with and something you will need to resolve on your own
> 
> What to do then?

Hey!
The only suggestion I could give you here is trying to play with the different `Chat_Listener_Priority:`.
Try changing the listener to HIGHEST or MONITOR (And don't forget to restart your server every time you change the Listener, and see if it works.

**Comment by kangarko** [comment]
Yeah that might be the last resort option, although I'm pretty sure it won't help, for now I'm marking this as wontfix, apologies about this.
